### DISCUSSION TITLE
On Click of any suggestion item `onChange` event of textarea is not getting fired.

### DISCUSSION THREAD

**Discussion by Sachin-chaurasiya** [github/markdown-toolbar-element#60] [Q&A]
I am using this library with reactjs and I have attached the onChange handler to capture the textarea changes and store them in the state. but whenever I click on any suggestion item it's get added to the text area but the change event is not firing up.
CC: @keithamus

**Comment by keithamus** [comment]
Are you able to bind an event listener without React? Do you have a reproduction in an environment I can see, for example Codepen?

**Reply by Sachin-chaurasiya** [reply]
@keithamus, here is the sandbox link [markdown editor](https://codesandbox.io/s/markdown-editor-lbfqi) .

**Reply by keithamus** [reply]
I manually bound an event listener to the DOM node using `$0.addEventListener('change', console.log)` and was able to see change events firing in the console. This looks like it might have something to do with how React handles events.

Have you tried changing the code to not use the `onChange` prop, and instead manually bind using `addEventListener` on the ref?

**Reply by Sachin-chaurasiya** [reply]
@keithamus , I haven't tried manually binding using addEventListener on the ref. I'll try and update the same here.

**Reply by Sachin-chaurasiya** [reply]
@keithamus, getting the same behaviour with the above mention approach.

**Reply by Sachin-chaurasiya** [reply]
@keithamus , here https://github.com/github/text-expander-element/blob/main/src/text-expander-element.ts#L130 I can see we are just setting the value and not dispatching the `change` event.

**Reply by keithamus** [reply]
Hmm, yeah this looks like a bug. Would you like to raise a PR fixing this?

**Reply by Sachin-chaurasiya** [reply]
I have raised the PR for the same https://github.com/github/text-expander-element/pull/27.
